### PR TITLE
Create schedule spine incorporating timezones and daylight time

### PR DESCRIPTION
**Pull request by fivetran-jamie**
**Are you a current Fivetran customer?** 

fivetran made PR

**What change(s) does this PR introduce?** 

- adjusts circleci testing to be a bit more thorough
- creates schedule spine model that incorporates timezones and daylight time stuff. down the line, when calculcating ticket metrics, we will determine which schedule to use based on the schedule_id it is associated with and what schedule (ie PT vs PST) is active when the ticket is active (wondering how to define this though, like do we pin the ticket on its creation date? what if a ticket is created a day before Daylight Savings starts or ends?)
- incorporates the schedule spine into the business hours models
- regenerate docs

**Does this PR introduce a breaking change?**

- [x] Yes (please provide breaking change details below.)
- [ ] No  (please provide explanation how the change is non breaking below.)

**Is this PR in response to a previously created Issue**

- [x] Yes, Issue https://github.com/fivetran/dbt_zendesk/issues/62
- [ ] No 

**How did you test the PR changes?** 

 
- [ ] CircleCi  
- [ ] Other (please provide additional testing details below)
testing locally

**Select which warehouse(s) were used to test the PR**
 
- [x] BigQuery
- [ ] Redshift
- [ ] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🤌 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/intermediate/int_zendesk__schedule_spine.sql [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    -- standard schedule (includes timezones without DT)
+    -- starts: when the last Daylight Savings ended
+    -- ends: when the next Daylight Savings starts
+    select 
+        time_zone,
+        standard_offset_minutes as offset_minutes,
+
+        -- last_daylight_end_utc is null for the first record of the time_zone's daylight time, or if the TZ doesn't use DT
+        coalesce(last_daylight_end_utc, cast('1970-01-01' as date)) as valid_from,
```
Comment (fivetran-joemarkiewicz) [inline models/intermediate/int_zendesk__schedule_spine.sql:53]: I noticed that for any record that isn't the first, the `valid_from` and previous `valid_until` are the exact same timestamp. Could this possibly cause issues when doing joins downstream? I wonder if we would need the `valid_from` to be like a second after the previous `valid_until` so there is no overlap. Thoughts?

File: models/intermediate/int_zendesk__schedule_spine.sql [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    -- ends: when this Daylight Savings ends
+    select 
+        time_zone,
+        -- Pacific Time is -8h during standard time and -7h during DT
+        standard_offset_minutes + daylight_offset_minutes as offset_minutes,
+        daylight_start_utc as valid_from,
+        daylight_end_utc as valid_until
+
+    from order_timezone_dt
+    where daylight_offset_minutes is not null
```
Comment (fivetran-joemarkiewicz) [inline models/intermediate/int_zendesk__schedule_spine.sql:73]: Not sure I am understanding this?

File: models/intermediate/int_zendesk__schedule_spine.sql [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        schedule.schedule_name,
+        schedule.start_time - split_timezones.offset_minutes as start_time_utc,
+        schedule.end_time - split_timezones.offset_minutes as end_time_utc,
+
+        -- we'll use these to determine which schedule version to associate tickets with
+        split_timezones.valid_from,
+        split_timezones.valid_until
+
+    from split_timezones
+    join schedule
```
Comment (fivetran-joemarkiewicz) [inline models/intermediate/int_zendesk__schedule_spine.sql:92]: Can we `left join` here just to be consistent with style.

**Review by fivetran-joemarkiewicz — COMMENTED** [review]
See my comments inline

File: dbt_project.yml [hunk 4]
```diff
@@ -8,15 +8,15 @@ models:
   zendesk:
     +schema: zendesk
     +materialized: table
-    agent_work_time:
-      +materialized: ephemeral
+    # agent_work_time:
+    #   +materialized: ephemeral
```
Comment (fivetran-jamie) [inline dbt_project.yml:12]: just for testing -- reminder to remove

**Comment by fivetran-jamie** [comment]
flagging that i didn't add changelog stuff since that appears to the be in the release branch already? or i will add stuff after this PR is merged there...

**Comment by fivetran-jamie** [comment]
reminder to change packages.yml

**Review by fivetran-joemarkiewicz — APPROVED** [review]
Looks good! Just be sure to add the CHANGELOG

**Pull request merged**